ui/imports/shared/popups/UserStatusContextMenu.qml

```diff
@@ -28,74 +25,55 @@ PopupMenu {
         icon: root.store.userProfileInst.icon
     }
 
-    Item {
-        height: root.topPadding
-    }
-
-    Separator {
+    StatusMenuSeparator {
     }
 
-    Action {
+    StatusMenuItem {
         text: qsTr("View My Profile")
-
-        icon.source: Style.svg("profile")
-        icon.width: 16
-        icon.height: 16
-
+        icon.name: "profile"
         onTriggered: {
             Global.openProfilePopup(root.store.userProfileInst.pubKey)
             root.close()
         }
     }
 
-    Separator {
+    StatusMenuSeparator {
     }
 
-    Action {
+    StatusMenuItem {
+        id: alwaysOnlineAction
         text: qsTr("Always online")
+        image.source: Style.svg("statuses/online")
+        image.width: 12
+        image.height: 12
+        fontSettings.bold: root.store.userProfileInst.currentUserStatus === Constants.currentUserStatus.alwaysOnline
         onTriggered: {
-            //TODO move this to the store as soon as #4274 is merged
-            if (userProfile.currentUserStatus !== Constants.currentUserStatus.alwaysOnline) {
-                mainModule.setCurrentUserStatus(Constants.currentUserStatus.alwaysOnline);
-            }
+            store.setCurrentUserStatus(Constants.currentUserStatus.alwaysOnline)
             root.close();
         }
-
-        icon.color: Style.current.green
-        icon.source: Style.svg("online")
-        icon.width: 16
-        icon.height: 16
     }
 
-    Action {
+    StatusMenuItem {
+        id: inactiveAction
         text: qsTr("Inactive")
+        image.source: Style.svg("statuses/inactive")
```

Comment:
We **can** do it this way, but we could also introduce these as icons and have something like `icon.name:"status-inactive"`

Just spitballing here :)